imap/xapian_wrap.cpp

```diff
@@ -2089,10 +2090,7 @@ xapian_query_new_match(const xapian_db_t *db, int partnum, const char *str)
         return NULL;
     }
 
-    const char *prefix = get_term_prefix(partnum);
-    if (!prefix) {
-        return NULL;
-    }
+    std::string(get_term_prefix(partnum));
```

Comment:
Something fishy here, what's this line do?  The only thing I can see is the side effect of `get_term_prefix()` asserting if partnum is out of range.  If that's the intention, a comment to explain it would be good, because otherwise it looks like dead code that just constructs and discards two std::strings.  It's actually surprising to me that there isn't a warning!

Reply:
Thanks, that was unintentional. I'm surprised there isn't a warning, too.